apps/src/templates/admin/MassDeleteContainer.tsx

```diff
@@ -0,0 +1,450 @@
+      console.log('No username data to convert');
+      return;
+    }
+
+    setIsConverting(true);
+    setConversionError(null);
+    console.log('Starting username to ID conversion...');
+
+    try {
+      const response = await fetch('/admin/convert_usernames_to_ids', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+          Accept: 'application/json',
+          'X-CSRF-Token': getCSRFToken(),
+        },
+        body: JSON.stringify({
+          csv_data: originalCsvData,
+          teacher_id: teacherId,
+        }),
+      });
+
+      if (!response.ok) {
+        console.info(
+          `HTTP error for convert_usernames_to_ids. status: ${response.status}`
+        );
+      }
+
+      const result = await response.json();
+
+      if (result.success) {
+        setProcessedCsvData(result.converted_data);
+        setCsvData(result.converted_data);
+        setHasUsernames(false);
+      }
+    } catch (error) {
+      console.error('Error during conversion:', error);
+      setConversionError(
+        error instanceof Error
+          ? error.message
+          : 'Failed to convert usernames to student IDs. Please try again.'
+      );
+    } finally {
+      setIsConverting(false);
+    }
+  };
+
+  const deleteProgress = async (isDryRun = true) => {
+    try {
+      const response = await fetch('/admin/delete_user_progress', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+          Accept: 'application/json',
+          'X-CSRF-Token': getCSRFToken(),
+        },
+        body: JSON.stringify({
+          csv_data: processedCsvData,
+          teacher_id: teacherId,
+          dry_run: isDryRun,
+        }),
+      });
+
+      if (!response.ok) {
+        const errorData = await response.json().catch(() => ({}));
+        throw new Error(
+          errorData.error || `HTTP error! status: ${response.status}`
+        );
+      }
+
+      const result = await response.json();
+
+      if (result.success) {
+        console.log('Delete operation completed successfully');
+        console.log('Result:', result);
+
+        // Show user-friendly feedback
+        alert(
+          `${
+            result.dry_run ? 'Test delete' : 'Actual delete'
+          } completed successfully. Check console for details.`
+        );
+      } else {
+        console.error('Delete operation failed:', result);
+        throw new Error(
+          result.error +
+            (result.details ? `\n\nDetails: ${result.details}` : '')
+        );
+      }
+    } catch (error) {
+      console.error('Error during deletion:', error);
+      alert(
+        error instanceof Error
+          ? `Delete operation failed: ${error.message}`
+          : 'Delete operation failed. Please try again.'
+      );
+      throw error;
+    }
+  };
+
+  const openingDirections = (
+    <div>
+      <h1>Mass Delete Student Progress</h1>
```

Comment:
What do you think about adding a link to our manual instructions so that they are easy to find from here?